ConsoleUserInterface.cs

```diff
@@ -1,42 +1,249 @@
+    {
+        byte[][] moveHistories = [xMoveHistory, oMoveHistory];
+
+        byte[] fixedXMoveHistory = new byte[moveHistories[0].Length];
+        byte[] fixedOMoveHistory = new byte[moveHistories[1].Length];
+
+        Array.Copy(xMoveHistory, fixedXMoveHistory, xMoveHistory.Length);
+        Array.Copy(oMoveHistory, fixedOMoveHistory, oMoveHistory.Length);
+
+        int turn = Tools.CurrentTurn(xMoveHistory, oMoveHistory);
+
+        while (true)
         {
-            if (Tools.IsElementOf(xSquares, i))
+            var pressedKey = Console.ReadKey().Key;
+            if (pressedKey == ConsoleKey.LeftArrow)
             {
-                gameState[i - 1] = Constants.xChar;
+                moveHistories = LeftKeyPress(turn, moveHistories[0], moveHistories[1]);
+
+                if (turn > 0) 
+                {
+                    turn--; 
+                }
+
+                Console.Clear();
+                ConsoleOutputs.ScrollHistoryGameDisplay(moveHistories[0], moveHistories[1]);
             }
-            else if (Tools.IsElementOf(oSquares, i))
+            else if (pressedKey == ConsoleKey.RightArrow)
             {
-                gameState[i - 1] = Constants.oChar;
+                moveHistories = RightKeyPress(turn, moveHistories[0], fixedXMoveHistory, moveHistories[1], fixedOMoveHistory);
+
+                if (turn < Constants.maximumNumberOfTurnsAndInputValues) 
+                { 
+                    turn++; 
+                }
+
+                Console.Clear();
+                ConsoleOutputs.ScrollHistoryGameDisplay(moveHistories[0], moveHistories[1]);
             }
-            else
+            else if (pressedKey == ConsoleKey.Enter)
             {
-                gameState[i - 1] = Constants.emptyChar;
+                break;
             }
         }
-        ConsoleOutputs.DisplayGameState(gameState);
     }
+
+    public static int ChooseGameHistoryFile(List<History> historyList)
+    {
+        ConsoleOutputs.DisplayLine(Constants.chooseAGameLine);
+        return ConsoleInputs.GetConsoleByteInput() - 1;
+    }
+
+    private static byte[][] RightKeyPress(int turn, byte[] xMoveHistory, byte[] fixedXMoveHistory, byte[] oMoveHistory, byte[] fixedOMoveHistory)
+    {
+        if (Tools.IsXTurn(turn))
+        {
+            xMoveHistory[TurnNumberToCorrectIndex(turn)] = fixedXMoveHistory[TurnNumberToCorrectIndex(turn)];
+        }
+        else
+        {
+            oMoveHistory[TurnNumberToCorrectIndex(turn)] = fixedOMoveHistory[TurnNumberToCorrectIndex(turn)];
+        }
+
+        return [xMoveHistory, oMoveHistory];
+    }
+
+    private static byte[][] LeftKeyPress(int turn, byte[] xMoveHistory, byte[] oMoveHistory)
+    {
+        if (Tools.IsXTurn(turn))
+        {
+            xMoveHistory[TurnNumberToCorrectIndex(turn)] = 0;
+        }
+        else 
+        {
+            oMoveHistory[TurnNumberToCorrectIndex(turn)] = 0;
+        }
+
+        return [xMoveHistory, oMoveHistory];
```

Comment:
[xMoveHistory, oMoveHistory] is used three times in one file. Seems like it wants its own data structure around it. Create a class that would encapsulate it